src/spreadinterp.cpp

```diff
@@ -78,8 +79,9 @@ static FINUFFT_ALWAYS_INLINE void set_kernel_args(
     FLT *args, FLT x, const finufft_spread_opts &opts) noexcept;
 static FINUFFT_ALWAYS_INLINE void evaluate_kernel_vector(
     FLT *ker, FLT *args, const finufft_spread_opts &opts) noexcept;
-template<uint8_t w, class simd_type = xsimd::make_sized_batch_t<
-                        FLT, find_optimal_simd_width<FLT, w>()>> // aka ns
+template<uint8_t w, uint8_t upsampfact,
```

Comment:
The only confusing thing is that in the rest of code we use double upsampfac == 2.0 or 1.25 as the switch, but here it's unit8 and 200 vs 125.  Would it be bad to template by upsampfac = 2.0 or 1.25 ?

Also, name upsampfact (does "t" mean "tag"?) is too close to upsampfac - suggest use double upsampfac as above.

Reply:
double templates are a feture of c++20 we are with c++17 so this is the only solution.